web/pf4/src/components/ChartWithLegend.tsx

```diff
@@ -94,30 +94,35 @@ class ChartWithLegend extends React.Component<Props, State> {
       normalizedOverlay = this.normalizeOverlay(overlayFactor);
     }
     const dataEvents: any[] = [];
+    let onClick: ((event: any) => void) | undefined = undefined;
     if (this.props.onClick) {
+      onClick = (event: any) => {
+        // We need to get coordinates relative to the SVG
+        const svg = event.target.viewportElement;
+        const pt = svg.createSVGPoint();
+        pt.x = event.clientX;
+        pt.y = event.clientY;
+        const clicked = pt.matrixTransform(svg.getScreenCTM().inverse());
+        let flatDP: VCDataPoint[] = this.props.data.flatMap<VCDataPoint>(line => line.datapoints);
+        if (showOverlay) {
+          flatDP = flatDP.concat(normalizedOverlay);
+        }
+        const closest = findClosestDatapoint(
+          flatDP,
+          clicked.x - padding.left,
+          clicked.y - padding.top,
+          this.state.width - padding.left - padding.right,
+          height - padding.top - padding.bottom);
+        if (closest) {
+          this.props.onClick!(closest);
+        }
+      };
+
       dataEvents.push({
         target: 'data',
         eventHandlers: {
           onClick: event => {
-            // We need to get coordinates relative to the SVG
-            const svg = event.target.viewportElement;
-            const pt = svg.createSVGPoint();
-            pt.x = event.clientX;
-            pt.y = event.clientY;
-            const clicked = pt.matrixTransform(svg.getScreenCTM().inverse());
-            let flatDP: VCDataPoint[] = this.props.data.flatMap<VCDataPoint>(line => line.datapoints);
-            if (showOverlay) {
-              flatDP = flatDP.concat(normalizedOverlay);
-            }
-            const closest = findClosestDatapoint(
-              flatDP,
-              clicked.x - padding.left,
-              clicked.y - padding.top,
-              this.state.width - padding.left - padding.right,
-              height - padding.top - padding.bottom);
-            if (closest) {
-              this.props.onClick!(closest);
-            }
+            onClick!(event);
```

Comment:
nitpicking... You probably don't need the bang.

Reply:
it's needed because we're in a lambda